docs/sdk/v3/guides/04-routing.md

```diff
@@ -3,4 +3,51 @@ id: routing
 title: Routing a Swap
 ---
 
-TODO
\ No newline at end of file
+## Introduction
+
+This guide will cover how to use the Uniswap Autorouter to compute optimal routes and execute swaps.
+It is based on the [Routing code example](https://github.com/Uniswap/examples/tree/main/v3-sdk/routing), found in the Uniswap code examples [repository](https://github.com/Uniswap/examples).
+To run this example, check out the guide's [README](https://github.com/Uniswap/examples/blob/main/v3-sdk/routing/README.md) and follow the setup instructions.
+
+:::info
+For a briefer on the SDK and to learn more about how these guides connect to the examples repository, please visit our [background](./01-background.md) page!
+:::
+
+In this example we will trade between **ETH and USDC**, but you can configure your example to us any two currencies and amount of input currency.
+
+The guide will **cover**:
+1. Getting a router instance
+2. Getting a route
+3. Swapping using a route
+
+At the end of the guide, we should be able to create a route and and execute a swap between any two currencies tokens using the example's included UI.
+
+## Getting a router instance
+
+To compute our route, we will use the `@uniswap/smart-order-router` package, specifically the `AlphaRouter` class which requires a chain ID and a provider. Note that routing is not supported for local forks, so we will use a mainnet provider even when swapping on a local fork.
+
+```typescript reference title="Instantiating an AlphaRouter" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/a8e82ac2ee543eed620f0750f35eb6860f9f2567/v3-sdk/routing/src/routing.ts#L23-L26
+```
+
+## Getting a route
+
+Next, we will create our options conforming to the `SwapOptionsSwapRouter02` interface, defining the wallet to use, slippage tolerance, and deadline for the transaction:
+
+```typescript reference title="Routing Options" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/e2ffa0e58ddd28443c60611c2fca8ab464aeb6eb/v3-sdk/routing/src/routing.ts#L29-L34
+```
+
+Using these options, we can now create a trade (`TradeType.EXACT_INPUT` or `TradeType.EXACT_OUTPUT`) with the currency and the input amount to use to get a quote. For this example, we'll use an `EXACT_INPUT` trade to get a quote outputted in the quote currency.
+
+```typescript reference title="Creating a route" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/e2ffa0e58ddd28443c60611c2fca8ab464aeb6eb/v3-sdk/routing/src/routing.ts#L36-L46
+```
+
+## Swapping using a route
+
+Using this route, we can now execute the trade using the route's computed calldata, values, and gas values.
+
+```typescript reference title="Using a route" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/e2ffa0e58ddd28443c60611c2fca8ab464aeb6eb/v3-sdk/routing/src/routing.ts#L60-L67
+```
```

Comment:
Should we do something along the lines "after executing the transaction, you should be able to observe..."